docs/connect-field-devices/opcua-driver.md

```diff
@@ -108,3 +122,50 @@ The following steps can be used to generate the keystore:
   * Make a connection attempt using OPC-UA driver, this will likely fail because the server does not trust client certificate.
   * After the failed connection attempt, the server should display the certificate used by the driver in the administration UI. The server UI should allow to set it as trusted.
   * Make another connection attempt once the certificate has been set to trusted, this connection attempt should succeed.
+
+## Substree Subscription
+
+The driver can be configured to recursively visit the children of folder node and create a Monitored Item for the value of each discovered variable node with a single channel in Asset configuration.
+
+> Warning: This feature should be used with care since it can cause high load on both the gateway and the server if the referenced folder contains a large number of nodes and/or the notification rate is high.
+
+### Channel configuration
+
+In order to configure the driver for performing the discovery operation, a single channel can be defined with the following configuration:
+
+* **type**: `READ`
+* **value.type**: `STRING` (see below)
+* **listen**: `true`
+* **node.id**: the node id of the root of the subtree to visit
+* **node.namespace.index**: the namespace index of the root of the subtree to visit
+* **node.id.type** the node id type of the root of the subtree to visit
+* **listen.subscribe.to.children** (**NEW**): `true`
+
+The rest of the configuration parameters can be specified in the same way as for the single node subscription use case.
+
+The **listen.sampling.interval**, **listen.queue.size** and **listen.discard.oldest** parameters of the root will be used for all subscriptions on the subtree.
+
+### Discovery procedure
+
+The driver will consider as folders to visit all nodes that whose type definition is `FolderType`, that is all nodes with the following reference:
+
+`HasTypeDefinition`:
+  * namespace index: 0
+  * node id: 61 (numeric)
+  * URN: `http://opcfoundation.org/UA/`
+
+The driver will subscribe to all variable nodes found.
+
+### Event reporting
+
+If the Driver is used by a Wire Asset, it will emit on the wire a single message per received event.
+
+All emitted events will contain a single property, the name of this property is the browse path of the source OPCUA node relative to the root folder defined in channel configuration or the node id, depending on the value of the **Subtree subscription events channel name format** global configuration parameter.
```

Comment:
> All emitted events will contain a single property, the name of this property is the browse path of the source OPCUA node relative to the root folder defined in channel configuration or the node id, depending on the value of the **Subtree subscription events channel name format** global configuration parameter.

All emitted events will contain a single property. Depending on the value of the **Subtree subscription events channel name format** global configuration parameter, the name of this property is the node id or the browsed path of the source OPCUA node relative to the root folder defined in the channel configuration.